### ISSUE TITLE
Document how to associate `sentry-trace` with manual instrumentation

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-docs#2984]
We need to document how to associate the `sentry-trace` if using manual instrumentation for Java, .NET, React Native and other SDKs if possible.

just a Java example:

```
// assume there's a span/transaction or just create a new one
final ISpan activeSpan = hub.getSpan();
final SentryTraceHeader sentryTraceHeader = activeSpan.toSentryTrace();

// assume you can intercept the request call and add its spans header
request.getHeaders().add(sentryTraceHeader.getName(), sentryTraceHeader.getValue());

activeSpan.finish();
```

cc @bruno-garcia @jennmueng @maciejwalkowiak @Tyrrrz @brustolin

**Comment by lobsterkatie** [comment]
@marandaneto Let's sync on this before you do it. My work on the dynamic sampling/tracestate headers will likely come into play.

**Comment by Tyrrrz** [comment]
.NET example:

```csharp
using var httpClient = new HttpClient(new SentryHttpMessageHandler());
```

I think this is pretty unique to .NET though.

One can also do this fully manually:

```csharp
var span = hub.GetSpan();
var traceHeader = span.GetTraceHeader();

using var request = new HttpRequestMessage(HttpMethod.Get, "...");
request.Headers.Add("sentry-trace", traceHeader.ToString());

span.Finish();
```

**Comment by marandaneto** [comment]
@maciejwalkowiak how would we do it on Java right now?

**Comment by maciejwalkowiak** [comment]
If the snippet is about creating a span around HTTP call, it think we should also show how to start a child span, set the status and finish. Like its done here: https://github.com/getsentry/sentry-java/blob/main/sentry-spring/src/main/java/io/sentry/spring/tracing/SentrySpanClientHttpRequestInterceptor.java#L38

**Comment by marandaneto** [comment]
that makes total sense, I just wanted to be sure that the important bits are correct which is:

```
final SentryTraceHeader sentryTraceHeader = activeSpan.toSentryTrace();
request.getHeaders().add(sentryTraceHeader.getName(), sentryTraceHeader.getValue());
```

**Comment by lobsterkatie** [comment]
FYI - IDK how soon it will be released, but once my dynamic sampling work on the `tracestate` header goes out, `toSentryTrace` will be deprecated in favor of `getTraceHeaders`, which will return both the `sentry-trace` and `tracestate` headers.

**Comment by crummy** [comment]
It's not clear to me how to instantiate a trace from a request with the SentryTraceHeader - is there any documentation on this?

**Comment by marandaneto** [comment]
@crummy indeed, thanks for reporting this, subscribe to this issue please https://github.com/getsentry/sentry-java/issues/1346

**Comment by marandaneto** [comment]
@lobsterkatie mind reviewing https://github.com/getsentry/sentry-docs/pull/3324 ? we've got questions from multiple people already so we need to tackle this at least for the Java SDK

**Comment by Olegt0rr** [comment]
Wanna implement manual passing trace_id for Python Sanic requests (sentry's integration do nothing here), but can't find any info about header name to pass trace_id

**Comment by marandaneto** [comment]
@Olegt0rr it's the PR's title, `sentry-trace`

**Comment by marandaneto** [comment]
Keeping it open as a placeholder for other platforms, it's still not documented for the other SDKs other than Java/Android afaik.

**Comment by marandaneto** [comment]
This should be common content and just the code snippet would change per platform, only relevant for server-side SDKs.
Java is already done but should adapt to the common content.

**Comment by ex-nerd** [comment]
Would be great to see some info about this in the `node` and `serverless (node)` docs. I've been digging into the code for days trying to figure out how to get traces to be passed from one lambda through into another when calling them via the AWS SDK instead of HTTP.

**Comment by callum-p** [comment]
:+1: me too. I've got the sentry trace-id header going from my SPA to graphql backend but unable to make use of it in the backend. I'm seeing it in my console.logs() but doing stuff like the below is doing nothing. Tried a bunch of different ways of using it besides that. There are no docs on how to do this properly.

```
Sentry.startTransaction({
        traceId: request.http.headers.get('sentry-trace')
})
```

**Comment by mgarf** [comment]
knowing how to get the current sentry-trace id would also be helpful. 

Once use case i have is for not rest calls

**Comment by JR-RomanZaiats** [comment]
> 👍 me too. I've got the sentry trace-id header going from my SPA to graphql backend but unable to make use of it in the backend. I'm seeing it in my console.logs() but doing stuff like the below is doing nothing (yes, the value is being retrieved correctly). Tried a bunch of different ways of using it besides that. There are no docs on how to do this properly.
> 
> ```
> context.transaction = Sentry.startTransaction({
>         op: 'gql',
>         name: request.operationName,
>         traceId: request.http.headers.get('sentry-trace')
>       })
> ```
> 
> Edit: actually finally got this working after taking a break and coming back and reading what the Express integration does.
> 
> ```
> import { extractTraceparentData } from '@sentry/tracing'
> ...
> const traceparentData = extractTraceparentData(request.http.headers.get('sentry-trace'))
> 
>       context.transaction = Sentry.startTransaction({
>         op: 'gql',
>         name: 'GraphQLTransaction',
>         ...traceparentData
>       })
> ```
> 
> Then it worked. Very easy, but some docs would have prevented hours of digging.



> 👍 me too. I've got the sentry trace-id header going from my SPA to graphql backend but unable to make use of it in the backend. I'm seeing it in my console.logs() but doing stuff like the below is doing nothing (yes, the value is being retrieved correctly). Tried a bunch of different ways of using it besides that. There are no docs on how to do this properly.
> 
> ```
> context.transaction = Sentry.startTransaction({
>         op: 'gql',
>         name: request.operationName,
>         traceId: request.http.headers.get('sentry-trace')
>       })
> ```
> 
> Edit: actually finally got this working after taking a break and coming back and reading what the Express integration does.
> 
> ```
> import { extractTraceparentData } from '@sentry/tracing'
> ...
> const traceparentData = extractTraceparentData(request.http.headers.get('sentry-trace'))
> 
>       context.transaction = Sentry.startTransaction({
>         op: 'gql',
>         name: 'GraphQLTransaction',
>         ...traceparentData
>       })
> ```
> 
> Then it worked. Very easy, but some docs would have prevented hours of digging.

Thanks bro! You saved my time! Would be good to have some examples of using custom instrumentation in the sentry docs

**Comment by chargome** [comment]
By now this should be documented in `<platform>/tracing/trace-propagation/custom-instrumentation/`
